phpunit/class-wp-theme-json-test.php

```diff
@@ -2071,4 +2071,27 @@ public function data_process_blocks_custom_css() {
 			),
 		);
 	}
+
+	public function test_internal_syntax_is_converted_to_css_variables() {
+		$result = new WP_Theme_JSON_Gutenberg(
+			array(
+				'version' => WP_Theme_JSON_Gutenberg::LATEST_SCHEMA,
+				'styles'  => array(
```

Comment:
In addition to block-level data, can we add test data for top-level, elements, and variations to cover the full space and recursion?

- top-level: `styles.color.background`
- block-level: `styles.blocks.core/post-terms.color.background`
- elements top-level: `styles.link.color.background`
- elements block-level: `styles.blocks.core/navigation.elements.link.color.background`
- style variations: `styles.blocks.core/quote.variations.plain.color.background`

Reply:
Addressed in https://github.com/WordPress/gutenberg/pull/50366/commits/e8b1197ee3ca7ab8e51d187465aa7db8d90166df and https://github.com/WordPress/gutenberg/pull/50366/commits/adc1de945dae0a4ecfecca4dac9031fed76e7ac0